pyoxidizer/src/py_packaging/distribution.rs

```diff
@@ -362,24 +475,142 @@ impl ParsedPythonDistribution {
-                .args(&["-m", "ensurepip"])
-                .status()
-                .expect("failed to run ensurepip");
+            warn!(logger, "{} doesnt exist", pip_path.display().to_string());
+            invoke_python(&python_paths, &logger, &["-m", "ensurepip"]);
         }
 
         pip_path
     }
 
+    /// Duplicate the python distribution, with distutils hacked
+    pub fn create_hacked_base(&self, logger: &slog::Logger) -> PythonPaths {
+        let venv_base = self.venv_base.clone();
+
+        let venv_dir_s = self.venv_base.display().to_string();
+
+        if !venv_base.exists() {
+            if self.os == "windows" {
+                let external_python = which("python").unwrap().clone();
+                warn!(logger, "python={}", external_python.display().to_string());
+                // TODO: check python version
+
+                let external_dist_prefix = external_python.parent().unwrap();
+
+                copy_dir(&external_dist_prefix, &venv_base).unwrap();
+
+                let external_dist_prefix_s = external_dist_prefix.display().to_string();
+                warn!(
+                    logger,
+                    "copied {} to create hacked base {}", external_dist_prefix_s, venv_dir_s
+                );
+            } else {
+                let dist_prefix = self.base_dir.join("python").join("install");
+
+                copy_dir(&dist_prefix, &venv_base).unwrap();
+
+                let dist_prefix_s = dist_prefix.display().to_string();
+                warn!(
+                    logger,
+                    "copied {} to create hacked base {}", dist_prefix_s, venv_dir_s
+                );
+            };
+        }
+
+        let python_paths = resolve_python_paths(&venv_base, &self.version);
+
+        invoke_python(&python_paths, &logger, &["-m", "ensurepip"]);
+
+        prepare_hacked_distutils(logger, &python_paths);
+
+        python_paths
+    }
+
+    /// Create a venv from the distribution at path.
+    pub fn create_venv(&self, logger: &slog::Logger, path: &PathBuf) -> PythonPaths {
+        let venv_dir_s = path.display().to_string();
+
+        // This will recreate it, if it was deleted
+        let python_paths = self.create_hacked_base(&logger);
+
+        if path.exists() {
+            warn!(logger, "re-using {} {}", "venv", venv_dir_s);
+        } else {
+            warn!(logger, "creating {} {}", "venv", venv_dir_s);
+            invoke_python(&python_paths, &logger, &["-m", "venv", venv_dir_s.as_str()]);
+        }
+
+        resolve_python_paths(&path, &self.version)
+    }
+
+    /// Create or re-use an existing venv
+    pub fn prepare_venv(
+        &self,
+        logger: &slog::Logger,
+        venv_path: Option<&String>,
+    ) -> (PythonPaths, HashMap<String, String>) {
+        let venv_dir_path = match venv_path {
+            Some(path_str) => PathBuf::from(path_str),
+            None => tempdir::TempDir::new("pyoxidizer-temp-venv")
```

Comment:
The `TempDir` instance needs to outlive the `Path`/`PathBuf` pointing to it because the directory will be deleted once the `TempDir` is dropped. i.e. this function needs to return an instance of `TempDir` somewhere. Probably the easiest way to do this is to include an `Option<TempDir>` in `PythonPaths`. I'll likely make this change as part of committing this...